### PR TITLE
Devise gem views on user and admin

### PR DESCRIPTION
**Pull request by curtydudes**
Story: This addresses the User Story #2 and part of #6 where the User once logged in will be routed to their own portfolio. 

image

Changes: 
1) There were some updates on the Gemfile as the Rspec-rails were updated to 5.0.0
2) Since I could not navigate freely around devise's paths and controller as I would like, I found a solution that would overwrite devise routes from having the user routed to the home/index page instead of the user portfolio. This was to hard code it in the applications controller. This also let me take control that in by logging out, the user is routed back to the log in page. 
3) I commented out the API as of the moment since it hinders the localhost from loading when it's uncommented. 
4) I created a portfolio controller, views and model to accommodate the routing of the user portfoilo.
5) I took out the validation with the user model because I have been running into trouble that even when I have already typed it in, the sign up page would alert me and say that it's empty and should be typed down. I will look into this in the future. 
6) In devise's registration form, I added entries for username, first name and last name which would help me navigate the user as it is also part of the schema. 
7) deleted the database.yml.sample as it was a duplicate and I already have a database.yml
8) in the shared/_links, generated by devise, I edited the link to for the log in session to route the user. this is a redundancy from #2 and this was one of my way of securing the code.
9) The last file was for the routes and have the portfolio be under a user so that when the user accesses the portfolio, they only see their own.

### CODE REVIEW

**Comment by curtydudes** [comment]
I accidentally deleted database.yml.sample. I restored it using git restore --source master config/database.yml.sample. 

Source: https://stackoverflow.com/questions/307579/how-do-i-copy-a-version-of-a-single-file-from-one-git-branch-to-another

**Comment by curtydudes** [comment]
I added the needed parameters in application controller as per the Devise documentation. This was the reason why the signup page would say that the username, first and last name were blank because they were unpermitted parameters. This way by adding them, I was able to uncomment the validation in the user model and in turn have the RSPEC model pass.

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+    #   # @bitcoin = @crypto.company_name
+    #   # @active = @client.stock_market_list(:mostactive)
+    #   # @sectors = @client.sectors('MARKET')
+    #   # # @all = client.crypto('SQL')
+    #   # # @market = @client.market
+    #   # # @spy = client.market['SPY'].close.price
+    #   # @fb = @client.price('FB')
+    #   # # @fb_logo = client.logo('FB')
+    #   # @chart = @client.chart('MSFT', 'id', chart_simplify: true)
+    #   # @logo = @client.logo('MSFT')
```
Comment (analyn-cajocson) [inline app/controllers/home_controller.rb]: > I commented out the API as of the moment since it hinders the localhost from loading when it's uncommented.

I think you have to delete this code and figure out how to improve this code's performance

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -42,7 +42,7 @@ class Users::RegistrationsController < Devise::RegistrationsController
 
   # If you have extra params to permit, append them to the sanitizer.
   # def configure_sign_up_params
-  #   devise_parameter_sanitizer.permit(:sign_up, keys: [:attribute])
+  #   devise_parameter_sanitizer.permit(:sign_up, keys: [:first_name])
```
Comment (analyn-cajocson) [inline app/controllers/users/registrations_controller.rb]: why did you change this code? hehe

File: app/models/user.rb [hunk 3]
```diff
@@ -5,4 +5,6 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   validates :username, presence: true
   validates :first_name, presence: true
+  validates :last_name, presence: true
+  # has_many :portfolios
```
Comment (analyn-cajocson) [inline app/models/user.rb]: Delete unnecesary comments

File: app/views/portfolios/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<%= csrf_meta_tags %>
+
+<h2>User Details</h2>
+
+ <h3><%= current_user.email%></h3>
+
+<%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
```
Comment (analyn-cajocson) [inline app/views/portfolios/index.html.erb]: add new line at the end of file

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (curtydudes) [reply app/controllers/home_controller.rb]: Okay maam. Got this. Will delete the code first.

File: app/controllers/users/registrations_controller.rb (hunk 2 above)
Comment (curtydudes) [reply app/controllers/users/registrations_controller.rb]: My bad, maam. This was one of my attempts with addressing the unpermitted parameters and I forgot to change this back

File: app/views/portfolios/index.html.erb (hunk 4 above)
Comment (curtydudes) [reply app/views/portfolios/index.html.erb]: Will do, maam. May I ask why it is a good practice to add new lines at the end of a file?

Comment (analyn-cajocson) [reply app/views/portfolios/index.html.erb]: To avoid Github's warning sign
image

**Review by analyn-cajocson — APPROVED** [review]
